### PR TITLE
SonarQube (batch 2)

### PR DESCRIPTION
**Pull request by jklap**
Fixes related to to running code analysis with SonarQube against Rails.

This is the second set of changes-- another set is being prepped for a PR in a day or two.

These three sets of changes address many of the bug/critical/high issues identified by SonarQube but not all (I can only handle doing these in batches ;)

### CODE REVIEW

File: src/main/java/rails/game/correct/CorrectionManager.java [hunk 1]
```diff
@@ -92,7 +92,8 @@ protected boolean execute(CorrectionAction action) {
     }
 
 
-    public boolean equals(CorrectionManager cm) {
+    public boolean equals(Object object) {
+        CorrectionManager cm = (CorrectionManager) object;
```
Comment (madoar) [inline src/main/java/rails/game/correct/CorrectionManager.java:96]: This is a dangerous assumption. 
You should add a type check here

Comment (jklap) [reply src/main/java/rails/game/correct/CorrectionManager.java:96]: Yes, yes it is. I was working off the assumption that any `equals()` in place were already invalid due to the parameter declaration of `CorrectionManager` but that would have actually caused the default `equals()` to be called vs the override. So yeah, this should be fixed.

Comment (jklap) [reply src/main/java/rails/game/correct/CorrectionManager.java:96]: LOL, this is what is in my current code base:
```
    public boolean equals(Object object) {
        if ( ! (object instanceof CorrectionManager) ) {
            return false;
        }
...
```

Evidently I already fixed it!

Comment (neutronc) [reply src/main/java/rails/game/correct/CorrectionManager.java:96]: As Madoar said please fix :)

Comment (jklap) [reply src/main/java/rails/game/correct/CorrectionManager.java:96]: I've already got this in the following commit-- I'll have to rebase and edit that commit to pull this out :(

I looked through the code and don't believe this `equals()` method is actually being executed and looking at code coverage from unit tests it is certainly not. Given that I was just going to sit on it for the next PR given the pain of rebasing.

**Pull request merged**